ocaml/xenopsd/xc/device_common.ml

```diff
@@ -528,7 +528,7 @@ let with_qmp_connection domid f =
 
     [qmp_send_cmd] can send optionally a file descriptor [send_fd] over the
     connection before it sends the command. This is required for some commands. *)
-let qmp_send_cmd ?send_fd domid cmd =
+let qmp_send_cmd ?send_fd ?may_fail domid cmd =
```

Comment:
you can do `?(may_fail=false)` to avoid having to match an optional boolean below.